share/revng/test/tests/daemon/test.py

```diff
@@ -86,20 +87,32 @@ async def client(pytestconfig: Config, request) -> AsyncGenerator[AsyncClientSes
             "-b",
             f"unix:{socket_path}",
         ],
-        stdout=PIPE,
+        stdout=log_file.fileno(),
         stderr=STDOUT,
-        text=True,
         env=new_env,
     )
 
+    def stop_daemon():
+        if process.returncode is not None:
+            return process.returncode
+
+        process.send_signal(signal.SIGINT)
+        try:
+            return process.wait(30.0)
```

Comment:
Will I have to wait 30 extra seconds before getting a crash?
These tests are already quite slow, waiting 30 more seconds would make me very sad.

You mentioned `.communicate` as the proper solution, is there anything stopping us from implementing the proper solution?

Reply:
> You mentioned .communicate as the proper solution, is there anything stopping us from implementing the proper solution?

This *is* a proper solution, we're using `TemporaryFile`, but it works well just as well. The problem with using `.communicate` is that it forces collection of stdout when terminating rather than allowing `termination` and `reading stdout` to happen at different times (which would look awkward and ugly in the code).

> Will I have to wait 30 extra seconds before getting a crash?

No, that's just a safeguard in case a similar situation shows up, basically:
* If the test runs smoothly -> instant termination
* If the test crashes -> instant termination
* If the test hangs at shutdown (such as with PIPE) -> termination happens 30 seconds late with `KILL` and the test fails because the exit code is `-9` as opposed to `0`

Reply:
Gotcha